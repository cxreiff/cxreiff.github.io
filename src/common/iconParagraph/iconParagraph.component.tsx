import { FC, ReactElement, ReactNode } from "react";

import styles from "./iconParagraph.module.scss";

type IconParagraphProps = {
  icon: ReactElement;
  children?: ReactNode;
};

export const IconParagraph: FC<IconParagraphProps> = ({ icon, children }) => (
  <p className={styles.paragraph}>
    <span className={styles.icon}>{icon}</span>
    <span className={styles.content}>{children}</span>
  </p>
);
