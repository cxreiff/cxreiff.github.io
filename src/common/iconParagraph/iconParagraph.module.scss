@use "../../shared.scss" as *;
@use "../../themed.scss" as *;

.paragraph {
  display: flex;

  .icon {
    margin: 0 0.6rem 0 0;
    align-self: center;

    @include themed() {
      color: t($color-secondary);
    }
  }

  .content {
    flex: 1;
  }
}
