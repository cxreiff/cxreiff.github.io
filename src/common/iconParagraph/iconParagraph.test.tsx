import { render } from '@testing-library/preact'
import { Moon } from 'lucide-preact'

import { IconParagraph } from './iconParagraph.component'

test('should render', () => {
    const { container, getByRole, getByText } = render(
        <IconParagraph icon={<Moon role={'img'} />}>
            TEST
        </IconParagraph>
    )

    expect(getByRole('img')).toBeInTheDocument()
    expect(getByText('TEST')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
})
