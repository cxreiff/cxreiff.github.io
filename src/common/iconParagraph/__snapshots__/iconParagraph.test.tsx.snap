// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <p
    class="paragraph"
  >
    <span
      class="icon"
    >
      <svg
        class="lucide lucide-moon"
        fill="none"
        height="24"
        role="img"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"
        />
      </svg>
    </span>
    <span
      class="content"
    >
      TEST
    </span>
  </p>
</div>
`;
